src/main/java/de/tum/in/www1/artemis/domain/quiz/QuizConfiguration.java

```diff
@@ -0,0 +1,109 @@
+package de.tum.in.www1.artemis.domain.quiz;
+
+import java.util.List;
+
+public interface QuizConfiguration {
+
+    /**
+     * Find the list of QuizQuestion of the implementor entity.
+     *
+     * @return the list of QuizQuestion belongs to the implementor entity
+     */
+    List<QuizQuestion> getQuizQuestions();
+
+    /**
+     * Set the parent of the given QuizQuestion to reconnect ignored JSON attributes.
+     *
+     * @param quizQuestion the QuizQuestion of which the parent to be set
+     */
+    void setQuestionParent(QuizQuestion quizQuestion);
+
+    /**
+     * Recreate missing pointers from children to parents that were removed by @JSONIgnore
+     */
+    default void reconnectJSONIgnoreAttributes() {// iterate through quizQuestions to add missing pointer back to quizExercise
```

Comment:
I know you just moved this code, but I think this is a good time to refactor this function. Please split up logic for the different instances into their own functions (potentially private static functions).

Reply:
code is refactored

Reply:
Nice!